Versioning and contribution documentation (#190)
trikoder/oauth2-bundle: CONTRIBUTING.md

# Contributing

All contributions are **welcome** and **very much appreciated**.

We accept contributions via Pull Requests on [Github](https://github.com/trikoder/oauth2-bundle).

## Pull Request guidelines

> - **Add tests!** - If you are contributing with code, your contribution won't be accepted if it doesn't have tests.

Reviewer: @toniperic Even though I completely agree with this, it just sounds a bit too harsh. We don't want to drive away potential contributors. Perhaps something in the line of "it is important to add tests that confirm your changes work as expected". I know this leaves a huge door open for people skipping on writing tests, but in my experience whoever is willing to create a PR is usually willing to do it right, which includes tests :smile:

- **Send coherent history** - Make sure each individual commit in your pull request is meaningful. If you had to make multiple intermediate commits while developing, please [squash them](http://www.git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Changing-Multiple-Commit-Messages) before submitting.

## Development

[Docker](https://www.docker.com/) 18.03+ and [Docker Compose](https://github.com/docker/compose) 1.13+ are required for the development environment.

### Building the environment

Make sure your Docker images are all built and up-to-date using the following command:

```sh
dev/bin/docker-compose build
```

> **NOTE:** You can target a different version of PHP during development by appending the `--build-arg PHP_VERSION=<version>` argument.

After that, install all the needed packages required to develop the project:

```sh
dev/bin/php composer install
```

### Debugging

You can run the debugger using the following command:

```sh
dev/bin/php-debug vendor/bin/phpunit
```

Make sure your IDE is setup properly, for more information check out the [dedicated documentation](docs/debugging.md).

### Code linting

> This bundle enforces the PSR-2 and Symfony code standards during development using the [PHP CS Fixer](https://cs.sensiolabs.org/) utility. Before committing any code, you can run the utility so it can fix any potential rule violations for you:

Reviewer: @toniperic 
```suggestion
This bundle enforces the PSR-2 and Symfony code standards during development by using the [PHP CS Fixer](https://cs.sensiolabs.org/) utility. Before committing any code, you can run the utility to fix any potential rule violations:
```
